### ISSUE TITLE
Problem with jobs being killed when using qsubcellfun

### ISSUE THREAD

**Issue by JosePMarques** [Donders-Institute/staff-scientists#20]
**Describe the issue**
I get non reproducible crashes of submitted jobs using qsubcellfun (while the majority runs successfully).
If I rerun the script using the same input data, a different subset of submitted jobs will appear as a crash.
The nodes where this crashes happen also doesn't seem to be consistent (a note might run sucessfully 4 jobs and fail 4 other jobs)

**Describe yourself**
1. José Marques
2. MR physicist
3. MR techniques

**Audience**
whoever is using the cluster....

**Test data**
/home/common/temporary/4Staff
**To Reproduce**
1. Open TestQsuberrors.m
2. run the script
3. wait 10 or 15 minutes

4. Look at the output image, you will see that some slices are missing...
5. if you rerun the script another set of slices will be missing.. wall time and memory are never an issue.
6. 

**Expected behavior**

**Screenshots**
![example](https://user-images.githubusercontent.com/19411903/195817041-31da24f3-4277-4665-b75a-d7c86673be32.jpg)

**Environment and versions (please complete the following information):**
using matlab on the cluster
**Additional context**

**Comment by schoffelen** [comment]
OK, I copied the data over (temporarily) to a more persistent location, and we will look into it, if needed.

Have you checked the error-logs?

**Comment by JosePMarques** [comment]
Hi, Andrey is already looking at it…

I looked at the error files, I think they were very close to: “computer says no…”
The .e????? files are just empty.
And the .o???? don’t show anything interesting
Memory was fine and so was wall time
And the crashing nodes were not consistent.

**Comment by JosePMarques** [comment]
My fault, when moving the data to the temporary folder I removed the data folder.
Line 17 should be
Input.data_path ='/'

**Comment by schoffelen** [comment]
OK thanks, it needed to be pwd indeed, I missed initially the fact that it was an attempt to load in the mat file from the current folder.

**Comment by schoffelen** [comment]
So far, it runs throught without error for me (I couldn't resist trying it out)

**Comment by achetverikov** [comment]
I was able to reproduce the error this afternoon, but it works fine if qsubfeval is used instead. The script also throws errors (when using qsubcellfun) pointing at error logs at `/var/spool/torque/mom_priv/jobs/` but when I try to open the logs, they aren't there anymore. I'll look more into it next week.

**Comment by achetverikov** [comment]
Actually, now it seems to run fine again. So: temporary problem with full scratch disk or too much simultaneous read-write?

**Comment by JosePMarques** [comment]
This "temporary" problem has been annoying me for the last 3 days! I will check if tonight everything still runs smoothly :)

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by marcelzwiers** [comment]
If it works, but not always, it then seems to me that the tg should be
involved in this...

_____________________
Sent from my phone

Op vr 14 okt. 2022 18:03 schreef José P. Marques ***@***.***>:

**Comment by JosePMarques** [comment]
Sorry, only today I had the chance to run this again.

It indeed run smoothly… the first time.

As I tried to repeat the process by just changing the regularisation parameter (the lambda)… suddenly the jobs start opening very slowly (one job every 5 seconds)…
and after claiming to have finished all the jobs, it never actually gives the results back to matlab.
***@***.***
(I actually counted the 64 and 108 jobs respectively  I had submitted on two different iterative matlab sessions… so all successfully completed)… and it is hanging there for the last 2 hours :S

Any idea what the problem can be? Some prioritization issue for having asked too many jobs or too much memory in one go?

José

**Comment by marcelzwiers** [comment]
For years it has been such that the jobid's on Torque may change / are not
in register with the jobid's that qsubcellfun has in memory. So qsubcellfun
waits forever for the e/o-files, whereas the jobs have completed and have
produced different e/o-files (unknown to qsubcellfun). The only thing you
can do is to restart qsubcellfun...

Op wo 19 okt. 2022 om 16:16 schreef José P. Marques <
***@***.***>:

**Comment by robertoostenveld** [comment]
@hurngchunlee can you confirm that the jobIDs on Torque may change / are not in register with the jobIDs that are printed on screen upon submission?

**Comment by hurngchunlee** [comment]
Yes, the `qstat` will keep all running and pending jobs; but completed jobs will only be kept for 1 hour (i.e. jobs completed one-hour ago will be gone from `qstat`).  This is to reduce the memory requirement and to have fast qstat response.

**Comment by robertoostenveld** [comment]
that is not what @marcelzwiers reports: he stated that job IDs _change_, not that they disappear (which might also be a reason for problems, but nevertheless differs).

**Comment by hurngchunlee** [comment]
sorry for misunderstanding the question ... no the job id should never change.

**Comment by marcelzwiers** [comment]
Well, it shouldn't but it happens. It was a long time ago that we looked
into it, and I don't remember the details, but I believe we thought it may
have to do with Torque re-running/scheduling the job. Perhaps qsubcellfun
would be more robust if it doesn't rely on the job id to retrieve the
results, but instead use the qsub option to write the e/o files with a name
determined by qsubcellfun?

Op do 20 okt. 2022 om 11:28 schreef Hurng-Chun Lee ***@***.***

**Comment by hurngchunlee** [comment]
Ok ... but it has nothing to do with changing JobID.

The issue Marcel mentioned is due to the fact that the -o and -e are pointed to a directory that doesn't exist at the job submission time; but the directory is later created as part of the job.

In this case, it confused the Torque as at the job submission time, the -o/-e are resolved as files with the specification as the prefix (because the directory doesn't exist); but at the end of the job, since the directory becomes available, it has to change the plan to write -o/-e into the directory (and in this case, it doesn't use jobName as the prefix; but the jobId as the prefix).

If the directory of -o/-e is available before job submission, it is never an issue like this.

**Comment by marcelzwiers** [comment]
No, that's a separate issue (well, not an issue actually), qsubcellfun
doesn't specify the e/o-files, i.e. they are written in the cwd.

Op do 20 okt. 2022 om 12:43 schreef Hurng-Chun Lee ***@***.***
